src/cart/crt_group.c

```diff
@@ -3370,6 +3370,43 @@ crt_register_eviction_cb(crt_eviction_cb cb, void *arg)
 	return rc;
 }
 
+int
+crt_register_event_cb(crt_event_cb event_handler, void *arg)
+{
+	struct crt_event_cb_priv	*cb_priv;
+	int				 rc = 0;
+
+	D_ALLOC_PTR(cb_priv);
+	if (cb_priv == NULL)
+		D_GOTO(out, rc = -DER_NOMEM);
+	cb_priv->cecp_func = event_handler;
+	cb_priv->cecp_args = arg;
+	D_RWLOCK_WRLOCK(&crt_plugin_gdata.cpg_event_rwlock);
+	d_list_add_tail(&cb_priv->cecp_link, &crt_plugin_gdata.cpg_event_cbs);
```

Comment:
we should probably traverse event list first to ensure same handler+arg tuple doesnt get resgisterred multiple times. doing so wont cause issues in cart, but might help caller avoid errors on their side